pyproject.toml

```diff
@@ -1,11 +1,85 @@
 [build-system]
 requires = [
-    "setuptools>=42",
-    "scikit-build~=0.18.0",
+    "scikit-build-core @ git+https://github.com/scikit-build/scikit-build-core",
     "Cython >=3.0.10, <3.1.0"
 ]
-build-backend = "backend"
-backend-path = ["_custom_build"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "RapidFuzz"
+dynamic = ["version"]
+requires-python = ">= 3.8"
+authors = [
+  {name = "Max Bachmann", email = "pypi@maxbachmann.de"},
+]
+description = "rapid fuzzy string matching"
+readme = "README.md"
+classifiers=[
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "License :: OSI Approved :: MIT License",
+]
+Homepage = "https://github.com/rapidfuzz/RapidFuzz"
+Documentation = "https://rapidfuzz.github.io/RapidFuzz/"
+Repository = "https://github.com/rapidfuzz/RapidFuzz.git"
+Issues = "https://github.com/rapidfuzz/RapidFuzz/issues"
+Changelog = "https://github.com/rapidfuzz/RapidFuzz/blob/main/CHANGELOG.rst"
+
+[project.optional-dependencies]
+all = [
+  "numpy"
+]
+
+[project.entry-points.pyinstaller40]
+hook-dirs = "rapidfuzz.__pyinstaller:get_hook_dirs"
+tests = "rapidfuzz.__pyinstaller:get_PyInstaller_tests"
+
+[tool.scikit-build]
+sdist.include = [
+  "src/rapidfuzz/*.cxx",
+  "src/rapidfuzz/distance/*.cxx",
+]
+sdist.exclude = [
+  ".github"
+]
+wheel.exclude = [
+  "**.pyx",
+  "**.cxx",
+  "**.pxd",
+  "**.cpp",
+  "**.hpp",
+  "**.h",
+  "CMakeLists.txt",
+  "generate.sh"
+]
+wheel.packages = ["src/rapidfuzz"]
+wheel.cmake = false
+
+[[tool.scikit-build.overrides]]
+if.any.system-cmake = ">=3.15"
+if.any.cmake-wheel = true
+wheel.cmake = true
+
+[[tool.scikit-build.overrides]]
+if.failed = true
+wheel.cmake = false
+
+[[tool.scikit-build.overrides]]
+if.any.env.CIBUILDWHEEL = true
+if.any.env.CONDA_BUILD = true
+if.any.env.PIWHEELS_BUILD = true
+if.any.env.RAPIDFUZZ_BUILD_EXTENSION = true
+wheel.cmake = true
+error-message = "failed to build C++ Extension in a packaged build"
```

Comment:
```suggestion
message.after-success = "{yellow}Failed to build C++ Extension in a packaged build, using Python fallback."
```

Reply:
FYI, an override that never matches won't be validated, so this likely never matched in your tests.

Reply:
The goal was that it would:
1) print a warning when falling back to the pure Python fallback in
```
[[tool.scikit-build.overrides]]
if.failed = true
wheel.cmake = false
```
so I guess this would have to be added there

2) in case the environment variables are set I actually want the build to fail completely + print an error message


I don't have any tests for this in github actions yet since this requires a platform where cmake wheels are not available. Do you know an easy way to test this in github actions?

Reply:
I misunderstood your correction  back then. It should certainly use `messages.*`. In which context would this color actually have an effect?

Reply:
This is a success / failure message. If the override matches and the pure Python wheel is built, it "succeeds". And the color is the same color system that scikit-build-core uses, it shows up with `pypa/build` directly and with buffered builders like pip if you set `FORCE_COLOR`.